pinot-core/src/main/java/com/linkedin/pinot/core/minion/SegmentPurger.java

```diff
@@ -67,6 +67,16 @@ public File purgeSegment() throws Exception {
     LOGGER.info("Start purging table: {}, segment: {}", tableName, segmentName);
 
     try (PurgeRecordReader purgeRecordReader = new PurgeRecordReader()) {
+      // Make a first pass through the data to see if records need to be purged or modified
+      while (purgeRecordReader.hasNext()) {
+        purgeRecordReader.next();
+      }
+
+      if(_numRecordsModified == 0 && _numRecordsPurged == 0) {
+        // Returns null if no records to be modified or purged
+        return null;
```

Comment:
Do we have a metric that tracks this condition?

Reply:
We don't. Actually this might happen pretty frequently because we will purge same segment per several days.

Reply:
I definitely agree that we should have a metric. We don't have any open source minion metrics right now, so it might be a little more involved. I filed a ticket for this change.